plugins/scaffolder-backend/src/service/router.ts

```diff
@@ -137,11 +154,24 @@ export type ActionPermissionRuleInput<
   TParams
 >;
 function isActionPermissionRuleInput(
-  permissionRule: TemplatePermissionRuleInput | ActionPermissionRuleInput,
+  permissionRule: ScaffolderPermissionRuleInput,
 ): permissionRule is ActionPermissionRuleInput {
   return permissionRule.resourceType === RESOURCE_TYPE_SCAFFOLDER_ACTION;
 }
 
+/**
+ *
+ * @public
+ */
+export type TaskPermissionRuleInput<
+  TParams extends PermissionRuleParams = PermissionRuleParams,
+> = PermissionRule<string, {}, typeof RESOURCE_TYPE_SCAFFOLDER_TASK, TParams>;
```

Comment:
this should be

```suggestion
> = PermissionRule<SerializedTask, {}, typeof RESOURCE_TYPE_SCAFFOLDER_TASK, TParams>;
```

or 

```suggestion
> = PermissionRule<TaskSpec, {}, typeof RESOURCE_TYPE_SCAFFOLDER_TASK, TParams>;
```

depending on what @backstage/scaffolder-maintainers recommend.

Reply:
Ah so I chose to use `string` instead since this rule was to check for user ownership. Since we had a filter in the `/tasks` request to grab all tasks owned by a specific user, I thought it would be more performant to just check if the user is allowed to use the specified `userEntityRef` for the endpoint filter to avoid an unnecessary database query.

Reply:
yeah this doesn't mean that we have to query the database again, the filtering logic is fine! 
But if we define `string` instead of `TaskSpec` here, we might have troubles in the future in case we introduce more rules. Let's say we'll introduce `hasMetadata` rule, we won't be able to access the entire `TaskSpec` but only the `owner` as string.